client/app/components/tags-control/TagsControl.jsx

```diff
@@ -52,11 +54,14 @@ export class TagsControl extends React.Component {
     return (
       <div className={"tags-control " + this.props.className} data-test="TagsControl">
         {this.props.children}
-        {map(this.props.tags, tag => (
+        {flatMap(this.props.tags, (tag, i) => [
           <span className="label label-tag" key={tag} title={tag} data-test="TagLabel">
             {tag}
-          </span>
-        ))}
+          </span>,
+          this.props.tagSeparator && i < this.props.tags.length - 1 && (
+            <span className="tag-separator">{this.props.tagSeparator}</span>
+          ),
+        ])}
```

Comment:
Consider desctructuring props (instead of using `this.props.***` everywhere) to make code more readable:
`const { tags, tagSeparator } = this.props;`

Suggestion for this code fragment:

```jsx
{map(tags, (tag, index) => (
  <React.Fragment key={tag}>
    {tagSeparator && (index > 0) && <span className="tag-separator">{tagSeparator}</span>}
    <span className="label label-tag" key={tag} title={tag} data-test="TagLabel">
      {tag}
    </span>
  </React.Fragment>
)}
```

Reply:
Nice suggestions.  Changes made.